### PR TITLE
gpinitsystem tests updates for parity with 6x

### PR DESCRIPTION
**Pull request by piyushc01**
Incorporating review comments from 6x PR 
https://github.com/greenplum-db/gpdb/pull/14506

Ran the sanity pipeline and gpinitsystem tests are green. 

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 1]
```diff
@@ -330,5 +330,5 @@ Feature: gpinitsystem tests
          Then gpinitsystem should return a return code of 0
          #Verify entries in the gp_segment_configuration as expected
          And verify that cluster config has host-name populated correctly
-         #restore hosts file
-         And the user runs command "sudo mv -f /etc/hosts_orig /etc/hosts; rm -f /tmp/clusterConfigFile-1; rm -f /tmp/hostfile--1"
\ No newline at end of file
+         #restore hosts file, cleanup of hostlist, config file
+         And restore /etc/hosts file and cleanup hostlist file
```
Comment (shrakesh) [inline gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:334]: add a new line at the end of file

**Review by shrakesh — APPROVED** [review]
LGTM

**Review by jnihal — APPROVED** [review]
LGTM

**Review by Annu149 — APPROVED** [review]
LGTM

**Pull request merged**